xarray/backends/api.py

```diff
@@ -22,17 +22,35 @@
 
 
 def _get_default_engine(path, allow_remote=False):
-    if allow_remote and is_remote_uri(path):  # pragma: no cover
+    if allow_remote and is_remote_uri(path):
         try:
             import netCDF4
             engine = 'netcdf4'
-        except ImportError:
+        except ImportError:  # pragma: no cover
             try:
                 import pydap  # flake8: noqa
                 engine = 'pydap'
             except ImportError:
                 raise ValueError('netCDF4 or pydap is required for accessing '
                                  'remote datasets via OPeNDAP')
+    elif is_grib_path(path):
```

Comment:
@shoyer intent is:
 1. if you have *PyNIO* installed use it by default
 2. if you don't, but have *cfgrib* installed suggest the correct option
 3. alert that you need a GRIB backend installed

This make the best user experience with no unexpected behaviour.

I used PyNIO as the default engine because this is the simplest way for me to test, but assuming cfgrib is in general easier to install (and we are working to make it easier with a wheel including ecCodes), I'd suggest setting *cfgrib* as the default.

Your call.